File: frontend/src/api/system-config.ts
// Copyright (c) 2026 Jiamin Zhang (zjm20@vip.qq.com)
// Licensed under the MIT License. See LICENSE file for details.

import client from './client';

export interface SystemConfig {
  id: number;
  config_key: string;
  config_value: string | null;
  description: string | null;
  updated_by: string | null;
  updated_at: string | null;
}

export async function getSystemConfig(key: string): Promise<SystemConfig> {
  const res = await client.get(`/system-config/${key}`);
  return res.data;
}

export async function updateSystemConfig(key: string, value: string): Promise<SystemConfig> {
  const res = await client.put(`/system-config/${key}`, { config_value: value });
  return res.data;
}

File: backend/app/models/system_config.py
# Copyright (c) 2026 Jiamin Zhang (zjm20@vip.qq.com)
# Licensed under the MIT License. See LICENSE file for details.

from datetime import datetime

from sqlalchemy import Column, DateTime, Integer, String, Text

from app.database import Base


class SystemConfig(Base):
    """系统配置表"""
    __tablename__ = "system_configs"

    id = Column(Integer, primary_key=True, autoincrement=True)
    config_key = Column(String(100), unique=True, nullable=False, comment="配置键")
    config_value = Column(Text, nullable=True, comment="配置值")
    description = Column(String(200), nullable=True, comment="配置描述")
    updated_by = Column(String(50), nullable=True, comment="最后更新人")
    updated_at = Column(DateTime, default=datetime.now, onupdate=datetime.now, comment="更新时间")